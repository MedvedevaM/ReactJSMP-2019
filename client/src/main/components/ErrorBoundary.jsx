import React, { Component } from 'react';

export default class ErrorBoundary extends Component {
    state = {
      hasError: false,
    }

    componentDidCatch() {
      this.setState({ hasError: true });
    }

    render() {
      const { hasError } = this.state;
      if (hasError) {
        return <h1 className="error">Something went wrong.</h1>;
      }
      return this.props.children;
    }
}
