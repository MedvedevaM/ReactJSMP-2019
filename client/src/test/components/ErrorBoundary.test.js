import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';
import ErrorBoundary from "../../main/components/ErrorBoundary.jsx";

it("renders correctly", () => {
    const error = shallow(<ErrorBoundary />);
    expect(toJson(error)).toMatchSnapshot();
});

it("has an error", () => {
    const error = shallow(<ErrorBoundary />);
    error.setState({ hasError: true });
    expect(toJson(error)).toMatchSnapshot();
});