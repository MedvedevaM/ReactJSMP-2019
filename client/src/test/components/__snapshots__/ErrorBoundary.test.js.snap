// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has an error 1`] = `
<h1
  className="error"
>
  Something went wrong.
</h1>
`;

exports[`renders correctly 1`] = `""`;
